plugin/kvictoria/kvictoria.go

```diff
@@ -0,0 +1,243 @@
+package kvictoria
+
+import (
+	"errors"
+	"net"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	vm "github.com/VictoriaMetrics/metrics"
+	"github.com/twmb/franz-go/pkg/kerr"
+	"github.com/twmb/franz-go/pkg/kgo"
+)
+
+var (
+	// interface checks to ensure we implement the hooks properly
+	_ kgo.HookNewClient           = new(Metrics)
+	_ kgo.HookClientClosed        = new(Metrics)
+	_ kgo.HookBrokerConnect       = new(Metrics)
+	_ kgo.HookBrokerDisconnect    = new(Metrics)
+	_ kgo.HookBrokerWrite         = new(Metrics)
+	_ kgo.HookBrokerRead          = new(Metrics)
+	_ kgo.HookBrokerE2E           = new(Metrics)
+	_ kgo.HookBrokerThrottle      = new(Metrics)
+	_ kgo.HookGroupManageError    = new(Metrics)
+	_ kgo.HookProduceBatchWritten = new(Metrics)
+	_ kgo.HookFetchBatchRead      = new(Metrics)
+)
+
+// Metrics provides metrics using the [VictoriaMetrics/metrics] library.
+//
+// [VictoriaMetrics/metrics]: https://github.com/VictoriaMetrics/metrics
+type Metrics struct {
+	cfg cfg
+
+	sets map[string]*vm.Set
+}
+
+// NewMetrics returns a new Metrics that tracks metrics under the given namespace.
+//
+// You can pass options to configure the metrics reporting. See [Opt] for all existing options.
+func NewMetrics(namespace string, opts ...Opt) *Metrics {
+	return &Metrics{
+		cfg:  newCfg(namespace, opts...),
+		sets: make(map[string]*vm.Set),
+	}
+}
+
+// OnNewClient implements the [kgo.HookNewClient] interface for metrics gathering.
+// This method is meant to be called by the hook system and not by the user.
+func (m *Metrics) OnNewClient(client *kgo.Client) {
+	clientID := client.OptValue(kgo.ClientID).(string)
+
+	set, ok := m.sets[clientID]
```

Comment:
This reads as if it's meant to support the ability for the same `Metrics` struct to be used in multiple `kgo.Client`'s. If that's the case, should this be guarded with a sync.Mutex, here and in OnClientClosed?